# Estimando $\beta$ de los activos

<img style="float: right; margin: 0px 0px 15px 15px;" src="https://upload.wikimedia.org/wikipedia/commons/4/48/Capital_market_line_of_CAPM.svg" width="400px" height="400px" />

En la clase anterior aprendimos 
- ¿qué es el CAPM?; 
- ¿cuáles son los supuestos sobre los que se funda el CAPM?;
- derivamos la fórmula del CAPM; y
- aprendimos como obtener la $\beta$ de un portafolio a partir de la $\beta$ de activos individuales.

En la clase de hoy estudiaremos una forma de estimar los $\beta$ de activos individuales.

**Objetivos:**
- Revisitar riesgo sistemático y no sistemático.
- Estudiar un método para estimar las $\beta$ de los activos.

*Referencia:*
- Notas del curso "Portfolio Selection and Risk Management", Rice University, disponible en Coursera.
- [Notas del curso "Financial Engineering", Columbia University](http://www.columbia.edu/~ks20/FE-Notes/FE-Notes-Sigman.html)
___

## <font color=green> Conclusiones de la [lectura](https://web.stanford.edu/~wfsharpe/art/parable/parable.htm). </font>

## 1. Riesgo sistemático y no sistemático.

Recordamos la fórmula de CAPM:

$$E[r_i]-r_f=\beta_i(E[r_M]-r_f),$$

donde $\beta_i=\frac{\sigma_{M,i}}{\sigma_M^2}$ y $\sigma_{M,i}$ es la covarianza del portafolio de mercado con el activo individual $i$.

Todas las anteriores son variables determinísiticas.

- ¿Qué pasa si usamos el CAPM como un modelo de rendimientos? Es decir,

$$r_i=r_f+\beta_i(r_M-r_f)+\epsilon_i,$$

donde $\epsilon_i$ es un término de error.

<font color=green> Despejando $\epsilon_i$, tenemos que: </font>

- $E[\epsilon_i]=0$, y
- $cov(\epsilon_i,r_M)=0$.

<font color=green> Ver en el tablero. </font>

Entonces, la varianza del activo $i$ es:

$$\sigma_i^2=\beta_i^2\sigma_M^2+var(\epsilon_i),$$

donde el primer término corresponde al riesgo sistemático (de mercado) y el segundo al riesgo idiosincrático.
___

## 2. Estimando $\beta$ para un activo.

- En el mercado real, la cantidad de activos es ENORME, y tratar de construir el portafolio de mercado sería una tarea grandiosa, pero poco realista para un analista financiero. 
- Por lo tanto, los índices de mercado han sido creados para intentar aproximar el portafolio de mercado.
- Dicho índice es un portafolio más pequeño que el de mercado, construido por los que se consideran los activos más dominantes, y que capturan la esencia de el portafolio de mercado.

- El índice de mercado más conocido es el Standard & Poor’s 500-stock index (S&P), compuesto de 500 activos.
- Un $\beta$ para un activo dado, se puede estimar utilizando el S&P en reemplazo de M, y utilizando datos históricos para ambos rendimientos (el del activo y el del S&P500).
- Por ejemplo, considere un activo $i$ para el cual queremos estimar su $\beta_i$.
- Este estimado se construye usando medias, varianzas y covarianzas muestrales como sigue:
 - Escogemos $N$ rendimientos históricos, tales como los reportados mensualmente hace tres años.
 - Para $k=1,2,\dots,N$, $r_{ik}$ y $r_{S\&Pk}$ denotan el $k-$ésimo valor muestral de los rendimietos.

Entonces

$$\hat{E[r_i]}=\frac{1}{N}\sum_{k=1}^{N}r_{ik}, \text{ y}$$

$$\hat{E[r_{S\&P}]}=\frac{1}{N}\sum_{k=1}^{N}r_{S\&Pk}.$$

Además, la varianza $\sigma_{S\&P}^2$ se estima como

$$\hat{\sigma_{S\&P}^2}=\frac{1}{N-1}\sum_{k=1}^{N}(r_{S\&Pk}-\hat{E[r_{S\&P}]})^2,$$

y la covarianza $\sigma_{S\&P,i}$

$$\hat{\sigma_{S\&P,i}}=\frac{1}{N-1}\sum_{k=1}^{N}(r_{S\&Pk}-\hat{E[r_{S\&P}]})(r_{ik}-\hat{E[r_i]})$$

Finalmente, 

$$\hat{\beta_i}=\frac{\hat{\sigma_{S\&P,i}}}{\hat{\sigma_{S\&P}^2}}.$$

### Ejemplo...

Entrar a yahoo finance, a la información de MSFT, AAPL y ^GSPC.

In [1]:
# Importar paquetes
import pandas as pd
import pandas_datareader.data as web

In [2]:
# Función para descargar precios de cierre ajustados:
def get_adj_closes(tickers, start_date=None, end_date=None, freq='m'):
    # Fecha inicio por defecto (start_date='2010-01-01') y fecha fin por defecto (end_date=today)
    # Descargamos DataFrame con todos los datos
    closes = web.YahooDailyReader(symbols=tickers, start=start_date, end=end_date, interval=freq).read()['Adj Close']
    # Se ordenan los índices de manera ascendente
    closes.sort_index(inplace=True)
    return closes

In [3]:
# Importar datos de AAPL, MSFT, y ^GSPC
closes = get_adj_closes(tickers=['AAPL', 'MSFT', '^GSPC'],
                        start_date='2015-10-30')

In [4]:
closes

Symbols           AAPL        MSFT        ^GSPC
Date                                           
2015-11-01   27.330675   49.393021  2080.409912
2015-12-01   24.422157   50.759815  2043.939941
2016-01-01   22.584578   50.402992  1940.239990
2016-02-01   22.433773   46.551182  1932.229980
2016-03-01   25.424799   50.893883  2059.739990
...                ...         ...          ...
2020-07-01  106.068756  204.515610  3271.120117
2020-08-01  128.807755  224.986130  3500.310059
2020-09-01  115.809998  210.330002  3363.000000
2020-10-01  115.320000  204.720001  3310.110107
2020-10-30  108.860001  202.470001  3269.959961

[61 rows x 3 columns]

In [5]:
# Obtener rendimientos mensuales
ret = closes.pct_change().dropna()

In [6]:
ret

Symbols         AAPL      MSFT     ^GSPC
Date                                    
2015-12-01 -0.106420  0.027672 -0.017530
2016-01-01 -0.075242 -0.007030 -0.050735
2016-02-01 -0.006677 -0.076420 -0.004128
2016-03-01  0.133327  0.093289  0.065991
2016-04-01 -0.139921 -0.097049  0.002699
2016-05-01  0.065287  0.062763  0.015325
2016-06-01 -0.036831 -0.027776  0.000911
2016-07-01  0.090063  0.107680  0.035610
2016-08-01  0.018137  0.013761 -0.001219
2016-09-01  0.071276  0.008684 -0.001234
2016-10-01  0.004334  0.040278 -0.019426
2016-11-01 -0.026599  0.005674  0.034175
2016-12-01  0.053336  0.038164  0.018201
2017-01-01  0.047747  0.040393  0.017884
2017-02-01  0.128883 -0.010363  0.037198
2017-03-01  0.053236  0.035625 -0.000389
2017-04-01 -0.000070  0.039478  0.009091
2017-05-01  0.063418  0.020158  0.011576
2017-06-01 -0.053322 -0.007373  0.004814
2017-07-01  0.032704  0.054693  0.019349
2017-08-01  0.102669  0.028473  0.000546
2017-09-01 -0.056554  0.001563  0.019303
2017-10-01  0.096808  0.116660  0.022188
2017-11-01  0.016623  0.011902  0.003720
2017-12-01 -0.011706  0.021381  0.034343
2018-01-01 -0.010636  0.110708  0.056179
2018-02-01  0.063848 -0.013051 -0.038947
2018-03-01 -0.054210 -0.022089 -0.026884
2018-04-01 -0.015020  0.024652  0.002719
2018-05-01  0.130764  0.056886  0.021608
2018-06-01 -0.005599  0.001997  0.004842
2018-07-01  0.027983  0.075753  0.036022
2018-08-01  0.196227  0.058918  0.030263
2018-09-01 -0.004825  0.022079  0.004294
2018-10-01 -0.030478 -0.066101 -0.069403
2018-11-01 -0.184045  0.038199  0.017859
2018-12-01 -0.113616 -0.080090 -0.091777
2019-01-01  0.055154  0.028158  0.078684
2019-02-01  0.040315  0.072776  0.029729
2019-03-01  0.101731  0.057250  0.017924
2019-04-01  0.056436  0.107343  0.039313
2019-05-01 -0.127573 -0.052986 -0.065778
2019-06-01  0.134873  0.087127  0.068930
2019-07-01  0.076395  0.017244  0.013128
2019-08-01 -0.020184  0.011668 -0.018092
2019-09-01  0.077038  0.011845  0.017181
2019-10-01  0.110685  0.031216  0.020432
2019-11-01  0.074329  0.055869  0.034047
2019-12-01  0.102083  0.045294  0.028590
2020-01-01  0.054010  0.079455 -0.001628
2020-02-01 -0.116798 -0.048288 -0.084110
2020-03-01 -0.067554 -0.023883 -0.125119
2020-04-01  0.155374  0.136326  0.126844
2020-05-01  0.082165  0.022543  0.045282
2020-06-01  0.150492  0.113652  0.018388
2020-07-01  0.165132  0.007371  0.055101
2020-08-01  0.214380  0.100093  0.070065
2020-09-01 -0.100908 -0.065142 -0.039228
2020-10-01 -0.004231 -0.026672 -0.015727
2020-10-30 -0.056018 -0.010991 -0.012130

In [7]:
# Obtener matriz de covarianza
cov = ret.cov()
cov

Symbols      AAPL      MSFT     ^GSPC
Symbols                              
AAPL     0.007664  0.002980  0.002376
MSFT     0.002980  0.002758  0.001533
^GSPC    0.002376  0.001533  0.001775

In [11]:
# Beta de Microsoft
beta_MSFT = cov.loc['MSFT', '^GSPC'] / cov.loc['^GSPC', '^GSPC']
round(beta_MSFT, 2)

0.86

In [13]:
# Beta de Apple
beta_AAPL = cov.loc['AAPL', '^GSPC'] / cov.loc['^GSPC', '^GSPC']
round(beta_AAPL, 2)

1.34

## 3. Ejercicios

Su portafolio de inversión consiste de $\$150,000$ invertidos en un solo activo. Suponga que la tasa de interés libre de riesgo es $3\%$. Este activo tiene un rendimiento esperado del $12\%$, y una volatilidad del $40\%$, y el portafolio de mercado tiene un rendimiento esperado del $10\%$ y una volatilidad del $18\%$. Bajo las suposiciones del CAPM, ¿cuál es la volatilidad de un mejor portafolio alternativo que tiene el mismo rendimiento que el activo?

A. $\sigma = 15.52\%$

**B. $\sigma = 23.14\%$**

C. $\sigma = 30.25\%$ 

D. $\sigma = 35.13\%$ 

Con los mismos datos del ejercicio anterior, y suponiendo que usted está conforme con el riesgo del activo, bajo las suposiciones de CAPM, ¿cuál sería un mejor rendimiento esperado que debería ganar?

A. $E[r] = 13.65\%$

B. $E[r] = 15.30\%$

**C. $E[r] = 18.56\%$**

D. $E[r] = 20.38\%$ 

Suponga que usted tiene invertidos $\$30,000$ en los siguientes cuatro activos:

| Activo        | Cantidad invertida | $\beta$ |
| ------------- | ------------------ | ------- |
| Activo A      | $\$5,000$          | 0.75    |
| Activo B      | $\$10,000$         | 1.10    |
| Activo C      | $\$8,000$          | 1.36    |
| Activo D      | $\$7,000$          | 1.88    |

La tasa libre de riesgo es del $2\%$ y el rendimiento esperado del portafolio de mercado es del $8\%$. Con base en el CAPM, ¿cuál es la $\beta$ del portafolio?

A. $\beta_P = 0.95$

B. $\beta_P = 1.19$

**C. $\beta_P = 1.29$**

D. $\beta_P = 1.62$

Considere la siguiente distribución de rendimientos:

| Estado de la economía | Probabilidad       | $R_A$   |
| --------------------- | ------------------ | ------- |
| Depresión             | 0.30               | 0.10    |
| Normal                | 0.50               | 0.15    |
| Expansión             | 0.20               | 0.25    |

Suponiendo que el CAPM es válido, que la volatilidad del portafolio de mercado es $\sigma_M=0.10$ y que la correlación del activo A y el portafolio de mercado es $\rho_{A,M}=0.9$, ¿cuál es el valor de $\beta_A$?

A. $\beta_A=0.25$

**B. $\beta_A=0.47$**

C. $\beta_A=0.55$

D. $\beta_A=1.15$

Un fondo de inversiones con $\beta=0.8$ tiene un rendimiento esperado del $16\%$. Si la tasa libre de riesgo es $4\%$ y el rendimiento esperado del portafolio de mercado es del $13\%$, ¿debería invertir en este fondo?

**A. Si.**

B. No.

Suponga que encontramos un activo el cual, con base en su precio al día de hoy, tiene un rendimiento esperado más bajo que lo que sugiere la línea de mercado de activos. Si el CAPM es válido, ¿cuál(es) de las siguientes proposiciones es (son) verdaderas?

A. Si el activo tiene un rendimiento esperado más bajo que el sugerido por la línea, significa que su precio es muy bajo.

**B. Si el activo tiene un rendimiento esperado más bajo que el sugerido por la línea, significa que su precio es muy alto.**

C. El activo se grafica por encima de la línea.

**D. El activo se grafica bajo la línea.**

# Anuncios parroquiales

## 1. Último quiz el martes 10 de Noviembre.
## 2. Revisar archivo Tarea7: para el martes 3 de Noviembre.
## 3. Martes 3 de Noviembre y Viernes 6 de Noviembre no hay clase.

<script>
  $(document).ready(function(){
    $('div.prompt').hide();
    $('div.back-to-top').hide();
    $('nav#menubar').hide();
    $('.breadcrumb').hide();
    $('.hidden-print').hide();
  });
</script>

<footer id="attribution" style="float:right; color:#808080; background:#fff;">
Created with Jupyter by Esteban Jiménez Rodríguez.
</footer>